cmd/notation/plugin.go

```diff
@@ -62,3 +108,169 @@ func listPlugins(command *cobra.Command) error {
 	}
 	return tw.Flush()
 }
+
+func installPlugin(command *cobra.Command, args []string, force bool) error {
+	if len(args) != 1 {
+		return errors.New("missing plugin package")
+	}
+
+	pluginPath := args[0]
+
+	var pluginBinary string
+	// if plugin contains a file path split and select the last element in the array
+	if strings.Contains(pluginPath, "/") {
+		pluginBinary = strings.Split(pluginPath, "/")[len(strings.Split(pluginPath, "/"))-1]
+	} else {
+		pluginBinary = pluginPath
+	}
+
+	// get plugin metadata
+	cmd := exec.Command("./"+pluginPath, "get-plugin-metadata")
+
+	output, err := cmd.Output()
+	if err != nil {
+		panic(err)
+	}
+
+	var newPlugin map[string]interface{}
+	err = json.Unmarshal([]byte(output), &newPlugin)
+	if err != nil {
+		return err
+	}
+
+	pluginName := newPlugin["name"].(string)
+	newPluginVersion := newPlugin["version"].(string)
```

Comment:
Ah, so instead of executing the cmd and using Unmarshal to get the data use the [proto.GetMetadataResponse](https://github.com/notaryproject/notation/blob/main/cmd/notation/plugin.go#L53) instead? I'll start working on that, thanks for the feedback. :)

Reply:
@priteshbandi, I got this working, but it also causes you to assume the plugin name. For the NewCLIManager to get the metadata it needs to know the directory name of the plugin, which happens to also be the plugin name. However, the argument for the install command is given the binary name that by naming convention includes `notation` with an optional file ext of `.exe` for the Windows binary. The easy work around is to split the prefix off and the file extension to get the name of the plugin but if the plugin for any reason doesn't follow naming standards the logic will fail. Do you see any way around that?

Reply:
Hey @priteshbandi, I tinkered with this implementation a bit and came up with the following:

You can use the GetMetadata method from the CLIPlugin type to get the metadata from the binary, but it needs to be in a directory structure that mimics the notation layout. 

For example, a binary called `notation-azure-kv.exe` would have to be put into a directory called `$binaryPath\azure-kv` to extract the metadata from the binary. Otherwise, you'll an error:

> ` CreateFile C:\Users\userName\downloads\azure-kv\notation-azure-kv.exe: The system cannot find the path specified.

(If there's a way to get around this please LMK)

With that said that logic can be used to get the metadata from an existing plugin within Notation's plugin directory. What do you think about using exec.Command to get the metadata from the plugin being installed and use the GetMetaData method to get information from an existing plugin, if it exists?

Reply:
I've got a working change to replace exec.Command with the GetMetadata method. Once I've replaced the semver logic as well I'll push the changes to the PR branch. :)

Reply:
Replaced the exe.Command logic with [GetMetadata ](https://github.com/notaryproject/notation/pull/617/commits/e4d7ef5f0b4d53f2e1a218350742fab9f09c21ff) from the plugin.